lib/licensed/sources/cocoapods.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+require "json"
+require "pathname"
+require "uri"
+require "cocoapods-core"
+
+module Licensed
+  module Sources
+    class Cocoapods < Source
+      def enabled?
+        return unless Licensed::Shell.tool_available?("pod")
+
+        config.pwd.join("Podfile").exist? && config.pwd.join("Podfile.lock").exist?
+      end
+
+      def enumerate_dependencies
+        pods.map { |pod|
+          metadata = dependency_metadata(pod)
+          path = dependency_path(pod.root_name)
+
+          Dependency.new(
+            path: path,
+            name: metadata["name"],
+            version: metadata["version"],
+            metadata: {
+              "type"      => Cocoapods.type,
+              "homepage"  => metadata["homepage"],
+              "summary"   => metadata["summary"],
+              "platforms" => metadata["platform"]
+            }
+          )
+        }
+      end
+
+      private
+
+      def pods
+        if targets.nil?
+          lockfile.dependencies
+        else
+          targets_to_validate = podfile.target_definition_list.filter { |t| targets.include?(t.label) }
+          if targets_to_validate.any?
+            targets_to_validate.map(&:dependencies).flatten
+          else
+            raise Licensed::Sources::Source::Error, "Unable to find any target in the Podfile matching the ones provided in the config."
+          end
+        end
+      end
+
+      def targets
+        config.dig("cocoapods", "targets")&.map { |t| "Pods-#{t}" }
+      end
+
+      def lockfile
+        @lockfile ||= Pod::Lockfile.from_file(config.pwd.join("Podfile.lock"))
+      end
+
+      def podfile
+        @podfile ||= Pod::Podfile.from_file(config.pwd.join("Podfile"))
+      end
+
+      def dependency_metadata(pod)
+        metadata = JSON.parse(Licensed::Shell.execute("pod", "spec", "cat", "--regex", "^#{pod.root_name}$"))
```

Comment:
I am hesitant about this and I would like to have your opinion:

This shell command fetches all the metadata for the package which is a good thing. But on the other side, it slows down the execution a lot since we need to execute it for every dependency of the project.  

Do you think the metadata is more important than the speed ?

Reply:
oh, also was there a specific reason to use the regex option rather than just passing the pods name to the CLI?

```suggestion
        metadata = JSON.parse(Licensed::Shell.execute("pod", "spec", "cat", pod.root_name))
```

Reply:
Ok thank you for your input! I will remove the CLI command in order the speedup the process. 
I you have time and you are interested, I think it would be better to implement your suggestion in a new PR. 

> oh, also was there a specific reason to use the regex option rather than just passing the pods name to the CLI?

The `pod spec cat` command returns an error when the provided pod name matches many pods. Here is an example:

<img width="1913" alt="CleanShot 2022-12-12 at 10 24 25@2x" src="https://user-images.githubusercontent.com/39057274/207084112-c06dba24-9962-476c-a01c-3ec38f012faf.png">

By adding `^pod-name$` the spec command will execute successfully